pkg/controller/releasemanager/plan/syncSLORules.go

```diff
@@ -0,0 +1,137 @@
+package plan
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/go-logr/logr"
+	picchuv1alpha1 "go.medium.engineering/picchu/pkg/apis/picchu/v1alpha1"
+	"go.medium.engineering/picchu/pkg/plan"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	rgNameRegex = regexp.MustCompile("[^a-zA-Z0-9]+")
+)
+
+type SyncSLORules struct {
+	App                         string
+	Namespace                   string
+	Labels                      map[string]string
+	ServiceLevelObjectiveLabels picchuv1alpha1.ServiceLevelObjectiveLabels
+	ServiceLevelObjectives      []*picchuv1alpha1.ServiceLevelObjective
+}
+
+type SLOConfig struct {
```

Comment:
I think it’s worth pulling this out into its own file, with a comment below that `SLOConfig` methods can be found next to the relevant syncer.

Reply:
Github hid this comment so I almost missed it but this was a good tip, did this.